monitor/sapmon.py

```diff
@@ -6,23 +6,67 @@
 #       (c) 2019        Microsoft Corp.
 #
 import pyhdb
+from datetime import datetime, timedelta, date
+import http.client as http_client
 import requests, json
 import sys, argparse
 import decimal
-from datetime import datetime, timedelta, date
 import hashlib, hmac, base64
-import logging, http.client as http_client
+import logging, logging.config
 import hashlib
 import re
 
 ###############################################################################
 
-PAYLOAD_VERSION              = "0.3"
+PAYLOAD_VERSION              = "0.4"
 STATE_FILE                   = "sapmon.state"
 INITIAL_LOADHISTORY_TIMESPAN = -(60 * 1)
 TIME_FORMAT_HANA             = "%Y-%m-%d %H:%M:%S.%f"
 TIME_FORMAT_LOG_ANALYTICS    = "%a, %d %b %Y %H:%M:%S GMT"
 TIMEOUT_HANA                 = 5
+DEFAULT_CONSOLE_LOG_LEVEL    = logging.INFO
+DEFAULT_FILE_LOG_LEVEL       = logging.INFO
+LOG_FILENAME                 = "sapmon.log"
```

Comment:
With each execution, it would be nice to have a unique name for the log.
Eg. we can add the current timestamp to the logname
```suggestion
LOG_TS= datetime.datetime.now()
LOG_FILENAME = "sapmon" + LOG_TS.strftime('%Y-%m-%d-%H%M%S') + ".log"
```

Reply:
Not a fan of this, since it will spam the directory (the monitor payload runs every minute => 1,440 log files/day).
We can debate making the log less verbose, so it will be easier to separate individual payload executions.